dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -13,7 +14,24 @@ def save
     model_id = generate_id
     model_data = params["ml_model"]
     return head :bad_request if model_data.nil? || model_data == ""
-    profanity_or_pii = ShareFiltering.find_failure(model_data.to_s, request.locale)
+    begin
+      profanity_or_pii = ShareFiltering.find_failure(
+        model_data.except(:trainedModel).to_s, request.locale
+      )
+    rescue OpenURI::HTTPError => share_filtering_error
+    end
+    if share_filtering_error
+      FirehoseClient.instance.put_record(
+        :analysis,
+        {
+          study: 'ai-ml',
+          study_group: 'pii-profanity-api',
+          event: 'share_filtering_error',
+          user_id: current_user&.id,
+          data_json: model_data.except(:trainedModel).to_json
+        }
+        )
+    end
```

Comment:
What exactly do we return to the client in case of the same error if it does still occur?  I suspect it would be best if we didn't return the whole Ruby exception and instead just returned a clean HTTP error?

Reply:
If there's a PII/profanity API error, we rescue the exception and the save will succeed. We don't return an error to the client in this case since the save still goes through, but the model will bypass the PII/profanity filters if there is an API error.

Reply:
Oh, that's good to know, and makes sense.  Could we add a comment that that is deliberate behaviour?

Reply:
Good call! Added.